native/src/main/java/io/ballerina/lib/ibm.ibmmq/utils/CommonUtils.java

```diff
@@ -0,0 +1,202 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.ibm.ibmmq.utils;
+
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQMessage;
+import com.ibm.mq.MQPropertyDescriptor;
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Optional;
+
+import static io.ballerina.lib.ibm.ibmmq.Constants.IBMMQ_ERROR;
+import static io.ballerina.lib.ibm.ibmmq.utils.ModuleUtils.getModule;
+
+/**
+ * {@code CommonUtils} contains the common utility functions for the Ballerina IBM MQ connector.
+ */
+public class CommonUtils {
+
+    public static final String BTOPIC = "TOPIC";
+    public static final String TOPIC_OBJECT = "TOPIC_OBJECT";
+
+    private static final String ERROR_DETAILS = "ErrorDetails";
+    private static final BString ERROR_REASON_CODE = StringUtils.fromString("reasonCode");
+    private static final BString ERROR_ERROR_CODE = StringUtils.fromString("errorCode");
+    private static final BString ERROR_COMPLETION_CODE = StringUtils.fromString("completionCode");
+    private static final BString MESSAGE_PAYLOAD = StringUtils.fromString("payload");
+    private static final BString MESSAGE_PROPERTIES = StringUtils.fromString("properties");
+    private static final BString MESSAGE_PROPERTY = StringUtils.fromString("property");
+    private static final String BPROPERTY = "Property";
+    private static final String BMESSAGE_NAME = "Message";
+    private static final BString PD_VERSION = StringUtils.fromString("version");
+    private static final BString PD_COPY_OPTIONS = StringUtils.fromString("copyOptions");
+    private static final BString PD_OPTIONS = StringUtils.fromString("options");
+    private static final BString PD_SUPPORT = StringUtils.fromString("support");
+    private static final BString PD_CONTEXT = StringUtils.fromString("context");
+    private static final BString PROPERTY_VALUE = StringUtils.fromString("value");
+    private static final BString PROPERTY_DESCRIPTOR = StringUtils.fromString("descriptor");
+
+
+    private static final MQPropertyDescriptor defaultPropertyDescriptor = new MQPropertyDescriptor();
```

Comment:
What is the reason to have this static field ?

Reply:
If the user has not added any property descriptors, used this as the input param rather than creating new objects for each property